app/app.go

```diff
@@ -129,17 +127,28 @@ func New(opts ...configOption) (*App, error) {
 	return app, nil
 }
 
-func getIntFromEnvIfAvailable(name string) (int, error) {
-	strValue := os.Getenv(name)
-	if strValue == "" {
-		return 0, nil
+func parseDurationTimeout(l *zap.SugaredLogger, flag string, deprecatedFlag string) (time.Duration, bool, error) {
+	if strValue, ok := os.LookupEnv(flag); ok {
+		d, err := time.ParseDuration(strValue)
+		if err != nil {
+			return 0, false, fmt.Errorf("failed to parse %s: %w", flag, err)
+		}
+
+		return d, true, nil
 	}
 
-	value, err := strconv.Atoi(strValue)
-	if err != nil {
-		return -1, err
+	l.Warnf("%s is deprecated, please consider moving to %s", deprecatedFlag, flag)
```

Comment:
should this only be logged if the deprecated flag is actually specified? i.e. move it to line 143?

Reply:
Good catch! Thanks for spotting this, updated!